addon/components/dropdown-action.ts

```diff
@@ -0,0 +1,18 @@
+import Component from '@glimmer/component';
+import { scheduleOnce } from '@ember/runloop';
+import { tryInvoke } from '@ember/utils';
```

Comment:
you should also be passing a TS interface for the component's args into the component class declaration, e.g. like this: https://github.com/Gavant/ember-floating-labels/blob/master/addon/components/fl-select.ts#L5

In this component i think your only arg will be an optional `action` arg, so
```ts
action?: (event: Event) => void;
```